src/components/tabs/tabs.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import injectSheet from 'react-jss';
+import cn from 'classnames';
+import styles from './tabs-styles';
+
+class Tabs extends Component {
+  injectContextProps(displayName) {
+    const { children: childrenFromProp, value, variant, singlePanel, onChange } = this.props;
+    let index = 0;
+
+    return React.Children.map(childrenFromProp, child => {
+      if (child.type.displayName !== displayName || !React.isValidElement(child)) {
+        return null;
+      }
+
+      const childrenWithContext = React.cloneElement(child, {
+        selected: value === index,
+        index,
+        variant,
+        singlePanel,
+        changeHandler: child.type.displayName === 'Jss(Tab)' ? onChange : null,
+      });
+
+      index += 1;
+
+      return childrenWithContext;
+    });
+  }
+
+  render() {
+    const { classes, className, variant } = this.props;
+    const tabs = this.injectContextProps('Jss(Tab)');
```

Comment:
Do we really want to do this? Feels like there should be a better solution that to connect to 'Jss(Tab)'